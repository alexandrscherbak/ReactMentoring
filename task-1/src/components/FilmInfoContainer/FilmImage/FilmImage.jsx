import React from 'react';

const FilmImage = ({imageUrl, title}) => (
	<React.Fragment>
		<img src={imageUrl} alt={title} />

		<style jsx>
			{`
				img {
					max-width: 100%;
				}
			`}
		</style>
	</React.Fragment>
);

export default FilmImage;
