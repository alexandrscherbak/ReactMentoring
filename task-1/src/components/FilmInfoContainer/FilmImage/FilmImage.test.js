import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FilmImage from './FilmImage.jsx';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

const props = {
	title: 'title',
	imageUrl: 'imageUrl',
};

function setup() {
	return mount(<FilmImage {...props}/>);
}

describe('FilmImage', () => {
	test('snapshot', () => {
		const component = renderer.create(
			<FilmImage {...props}/>,
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should render self and subcomponents', () => {
		const enzymeWrapper = setup();
		expect(enzymeWrapper.find('img').html()).toContain(props.title);
	});
});
