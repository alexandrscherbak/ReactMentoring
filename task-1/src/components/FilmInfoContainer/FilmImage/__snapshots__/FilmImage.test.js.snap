// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmImage snapshot 1`] = `
<img
  alt="title"
  className="jsx-2994430438"
  src="imageUrl"
/>
`;
